src/utils/buildMetadata.ts

```diff
@@ -166,13 +153,198 @@ export function buildMetadata(sheets: Sheet[], defaultCC: string) {
         optionSets,
         trackedEntityAttributes,
         trackedEntityTypes,
-        //programs,
+        programSections: buildprogramSections(sheets),
+        programs: buildPrograms(sheets),
+        programStages: buildProgramStages(sheets),
+        programStageSections: buildProgramStageSections(sheets),
         programRules: buildProgramRules(sheets),
         programRuleActions: buildProgramRuleActions(sheets),
         programRuleVariables: buildProgramRuleVariables(sheets),
     };
 }
 
+function buildPrograms(sheets: Sheet[]) {
+    const get = (name: string) => getItems(sheets, name);
+
+    const programs = get("programs");
+    const pSections = get("programSections");
+    const pStages = get("programStages");
+    const trackedEntityTypes = get("trackedEntityTypes");
+    const categoryCombos = get("categoryCombos");
+
+    return programs.map(program => {
+        const trackedEntityType = {
+            id: getByName(trackedEntityTypes, program.trackedEntityType)?.id
+        };
+
+        const programStages = pStages.filter((programStages) => {
+            return programStages?.program === program.name;
+        }).map(programStages => ({ id: programStages.id }));
+
+        replaceById(program, "categoryCombo", categoryCombos);
+
+        const programSections = pSections.filter((programSections) => {
+            return programSections?.program === program.name;
+        }).map(programSections => ({ id: programSections.id }));
+
+        if (trackedEntityType.id) {
+            const programType = "WITH_REGISTRATION";
+            return { ...program, programType, trackedEntityType, programStages, programSections };
+        } else {
+            const programType = "WITHOUT_REGISTRATION";
+            return { ...program, programType, programStages };
+        }
+    });
+}
+
+function buildprogramSections(sheets: Sheet[]) {
+    const get = (name: string) => getItems(sheets, name);
+
+    const programSections = get("programSections");
+    const programs = get("programs");
+    const teAttributes = get("trackedEntityAttributes");
+
+    const sectionsAttributes = get("programSectionsTrackedEntityAttributes").map(trackedEntityAttributes => {
+        const programSection = programSections.find(
+            programSection => {
+                return programSection.name === trackedEntityAttributes.programSection &&
+                    programSection.program === trackedEntityAttributes.program
+            }
+        )?.id;
+
+        const trackedEntityAttribute = getByName(teAttributes, trackedEntityAttributes.name)?.id;
+
+        return { programSection, trackedEntityAttribute }
+    });
+
+    return programSections.map(programSection => {
+        const program = {
+            id: getByName(programs, programSection.program)?.id
+        };
+
+        if (typeof programSection.sortOrder === 'undefined') {
```

Comment:
Why is there this check? The `programSection` should not have a field called `sortOrder`, or how can it?

Reply:
The `sortOrder` field gets updated on batches of elements that belong to the same sortOrder pool, therefore the check to avoid processing the same elements multiple times.

Reply:
Since you are not going to modify the original `programSection`s, I understand that that is no longer the case (so you won't need that check). Is that right?